os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -437,6 +509,25 @@ trwifi_result_e wifi_netmgr_utils_connect_ap(struct netdev *dev, trwifi_ap_confi
 		}
 	}
 
+	ap_channel = 0xffff;
+
+	if (scan_number) {
+		int i;
+		rtw_cancel_timer(&(scan_timer));
+		for (i = 0 ; i < scan_number ; i++) {
+			if(strncmp(saved_scan_list[i].ssid, ap_connect_config->ssid, ap_connect_config->ssid_length) == 0) {
+				ap_connect_config->ap_auth_type = saved_scan_list[i].ap_auth_type;

```

Comment:
An application might pass ap_auth_type and crypto_type. according to this, driver overwrite application value. I think this is wrong. So you have to compare ap_auth type and crypto type too.

Reply:
@pillip8282, Does that mean we only search the full scan list for channel information to set the partial scan to improve the performance?
In this case, how do we handle the case of modification of channel before the expiry of scan duration(refresh of scan list)? Because it will set the partial scan with previous channel and results in wifi_connect failure. 

**Suggested solution:** since we only pick the channel from scan list, we can remove that particular entry in case of wifi_connect failure. So that for the second call from application, it will skip the partial scan part and does the normal wifi connection.

Reply:
> An application might pass ap_auth_type and crypto_type. according to this, driver overwrite application value. I think this is wrong. So you have to compare ap_auth type and crypto type too.

Modified as suggested, Please help to review.